### PR TITLE
Fixes and nit-picks

### PR DESCRIPTION
**Pull request by luanpotter**
This fixes a lot of warnings and small problems throughout the code.

Let's always keep it clean! :)

### CODE REVIEW

File: doc/examples/debug/lib/main.dart [hunk 1]
```diff
@@ -54,16 +57,16 @@ class MyGame extends BaseGame {
...
     android.x = 100;
     android.y = 400;
 
-    AndroidComponent android2 = AndroidComponent();
+    final AndroidComponent android2 = AndroidComponent();
     android2.x = 100;
     android2.y = 400;
     android2.yDirection = -1;
 
-    AndroidComponent android3 = AndroidComponent();
```
Comment (erickzanardo) [inline doc/examples/debug/lib/main.dart]: The type here is not necessary (I know, it was me who originally wrote this)

File: doc/examples/flare/lib/main.dart [hunk 2]
```diff
@@ -20,10 +20,11 @@ void main() {
 }
 
 class MyGame extends BaseGame {
-  final TextConfig fpsTextConfig = TextConfig(color: const Color(0xFFFFFFFF));
+  final TextConfig fpsTextConfig =
+      const TextConfig(color: const Color(0xFFFFFFFF));
 
-  final paint = Paint()..color = Color(0xFFe5e5e5e5);
-  List<String> _animations = ["Stand", "Wave", "Jump", "Dance"];
+  final paint = Paint()..color = const Color(0xFFe5e5e5e5);
```
Comment (erickzanardo) [inline doc/examples/flare/lib/main.dart]: Since you changed this line, make the color upper case

File: doc/examples/debug/lib/main.dart (hunk 1 above)
Comment (luanpotter) [reply doc/examples/debug/lib/main.dart]: I don't mind the type, it makes it more well defined. It's the missing final that the linter complains ;)

File: doc/examples/flare/lib/main.dart (hunk 2 above)
Comment (luanpotter) [reply doc/examples/flare/lib/main.dart]: that should be on our linter!!!

File: doc/examples/debug/lib/main.dart (hunk 1 above)
Comment (erickzanardo) [reply doc/examples/debug/lib/main.dart]: I realize that, but the type here is totally unnecessary, I am all for setting the type on method signatures, but on local attributions, where it is pretty clear what the type is, I say that we let the type inference work and we have less code written. But this is up to you.

**Pull request merged**